### PR TITLE
Origin 361 tests and helper methods

### PR DESCRIPTION
**Pull request by frisbee09**
Also built the documentation to merge into staging here

### CODE REVIEW

File: src/origin_sdk/service/Project.py [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
 from typing import Optional
-from tempfile import TemporaryDirectory
-import pandas as pd
-from io import StringIO
 
 logger = logging.getLogger(__name__)
 
 
 class Project:
+    """The Project class is a more user-friendly set of mappings around the
```
Comment (OscarPeace) [inline src/origin_sdk/service/Project.py:9]: Are these comments for us or end users? I think this is more detail than end users would want.

File: src/origin_sdk/service/Project.py [hunk 2]
```diff
@@ -20,10 +20,90 @@ def get(self, key: str):
...
+        return self
 
     def unpin(self):
+        """Unpins the project, if it isn't already"""
         if self.project.get("isProjectPinned"):
             self.session.unpin_project(self.project_id)
+        return self
+
+    def get_or_create_scenario_by_name(
+        self, scenario_name: str, base_scenario_id: Optional[str] = None
```
Comment (OscarPeace) [inline src/origin_sdk/service/Project.py:41]: I think this should be two methods. I understand it's like an 'upsert' equivalent but I think the optional base_scenario_id parameter makes it basically 2 functions

File: src/origin_sdk/service/Project.py [hunk 3]
```diff
@@ -20,10 +20,90 @@ def get(self, key: str):
...
+                    f"{self.project.get('name')}",
+                    f"' with name '{scenario_name}'",
+                )
+            )
+
+        return scenario
+
+    @staticmethod
+    def get_or_create_project_by_name(
+        session: OriginSession, name: str, pin_project: bool = False
```
Comment (OscarPeace) [inline src/origin_sdk/service/Project.py:86]: What Team/Product are these being created under?

Comment (OscarPeace) [reply src/origin_sdk/service/Project.py:86]: Also I would have expected two functions again

Comment (frisbee09) [reply src/origin_sdk/service/Project.py:86]: Customer's don't specify teams or products. I will add as a separate optional field.

Re: `get_or_create` it's a useful and I think common(?) pattern but I can read around this more and see if it's confirmation bias.

File: src/origin_sdk/service/Project.py (hunk 1 above)
Comment (frisbee09) [reply src/origin_sdk/service/Project.py:9]: Both

**Review by OscarPeace — APPROVED** [review]

**Pull request merged**